### ISSUE TITLE
Invalid call to lazypipe().pipe()/ Question: How to use lazypipe with filters?

### ISSUE THREAD

**Issue by 360disrupt** [OverZealous/lazypipe#23]
Is there a way to use `lazypipe` with `gulp-filter`?

# Code
```
var gulp = require('gulp');
var $ = require('gulp-load-plugins')();
var paths = gulp.paths;
var lazypipe = require('lazypipe')


var cssFilter = $.filter('**/*.css', {restore: true});
var endStream  =  function() {
  this.end();
}
cssFilter.on("error", endStream);


exports.duplicateApp = lazypipe()
    .pipe(cssFilter)
    .pipe($.replace, ['#0492ff', '#008787'])
    .pipe(cssFilter.restore)
```
The problem is related to that line
```
var cssFilter = $.filter('**/*.css', {restore: true});
```
it needs to be
```
.pipe($.filter,['**/*.css', {restore: true}])
```
But is there a way to write this differently, to get it working with lazypipe and still having the listeners/ restore function?

I thought about something similiar to this:
```
exports.duplicateApp = lazypipe()
    .pipe(function(){cssFilter = $.filter; cssFilter.on("error", endStream); return cssFilter;}, ['**/*.css', {restore: true}])
    .pipe($.replace('#0492ff', '#008787'))
    .pipe(function(){return cssFilter.restore;})
```
But I 1) run into the same error and 2) cssFilter would not be available in the other pipe.

**Comment by OverZealous** [comment]
I don't know anything about gulp-filter, but at a guess, your first solution might work like this:

```js
exports.duplicateApp = lazypipe()
    .pipe(function() { return cssFilter })
    .pipe($.replace, ['#0492ff', '#008787'])
    .pipe(function() { return cssFilter.restore })
```

That being said, it probably still won't be of any use, because you are storing the filter outside the function, which means you can't reuse the lazypipe (which is the entire point).  You'd probably be better off just wrapping the gulp-filter setup in your own function, and using `stream-combiner` on your own.

The thing to keep in mind is normal gulp `.pipe()`s take the output of the pipe-creation function (e.g., `«gulp.src».pipe(gulpConcat('somefile.js'))`, while lazypipe requires a function that creates a pipe, so it can lazily be called later when it's actually used (e.g.: `«lazypipe».pipe(gulpConcat, 'somefile.js')` will get evaluated when the pipe is executed).

Basically, you always have to pass a function into lazypipe's `pipe` function.

**Comment by OverZealous** [comment]
I haven't heard from you, so I will assume you solved this issue.

**Comment by 360disrupt** [comment]
Thanks, yes I packed it into functions.
